trieste/acquisition/function/function.py

```diff
@@ -689,3 +689,108 @@ def __call__(self, x: TensorType) -> TensorType:
+    space given a batch of query points.
+    """
+
+    def __init__(self, integration_points: TensorType, threshold: Sequence[float] = None) -> None:
+        """
+        :param integration_points: set of points to integrate the prediction variance over.
+        """
+        self._integration_points = integration_points
+        self._threshold = threshold
+
+    def __repr__(self) -> str:
+        """"""
+        return f"IntegratedVarianceReduction(threshold={self._threshold!r})"
+
+    def prepare_acquisition_function(
+        self,
+        model: ProbabilisticModel,
+        dataset: Optional[Dataset] = None,
+    ) -> AcquisitionFunction:
+        """
+        :param model: The model.
+        :param dataset: Unused.
+
+        :return: The determinant of the predictive function.
+        """
+
+        return integrated_variance_reduction(model, self._integration_points, self._threshold)
+
+    def update_acquisition_function(
+        self,
+        function: AcquisitionFunction,
+        model: ProbabilisticModel,
+        dataset: Optional[Dataset] = None,
+    ) -> AcquisitionFunction:
+        """
+        :param function: The acquisition function to update.
+        :param model: The model.
+        :param dataset: Unused.
+        """
+        return function  # no need to update anything
+
+
+class integrated_variance_reduction(AcquisitionFunctionClass):
+    """
+    The reduction of the average of the predicted variance over the integration points
+    (a.k.a. IMSE criterion). See :cite:`Picheny2010` for details.
+
+    If no threshold is provided, the original variance is used. Otherwise,
+    the variance is weighted by the posterior GP pdf evaluated at the threshold
+    (if a single value is given) or by the probability that the GP posterior belongs
+    to the interval between the 2 thresholds.
+    """
+
+    def __init__(
+        self,
+        model: ProbabilisticModel,
+        integration_points: TensorType,
+        threshold: Sequence[float] = None,
+    ):
+        """
+        :param model: The model of the objective function.
+        :param integration_points: points over which to integrate the objective prediction variance
+        :param threshold:
+        """
+        if not hasattr(model, "conditional_predict_f"):
+            raise AttributeError(
+                """
+                Integrated variance reduction only supports models with a
+                conditional_predict_f method.
+                """
+            )
+
+        self._model = model
+        self._integration_points = integration_points
+        if threshold is None:
+            self._weights = tf.cast(1.0, integration_points.dtype)
+        else:
+            mean_old, var_old = self._model.predict(query_points=integration_points)
+            distr = tfp.distributions.Normal(mean_old, tf.sqrt(var_old))
+            self._weights = distr.cdf(tf.cast(threshold[1], mean_old.dtype)) - distr.cdf(
```

Comment:
this means threshold is either None or a range,shouldn't there be an option for a single threshold, a scalar?

Reply:
added the option